test/lib/scsi/scsi_bdev/scsi_bdev_ut.c

```diff
@@ -451,6 +458,37 @@ inquiry_overflow_test(void)
 	}
 }
 
+/*
+ * This test is to verify specific error translation from bdev to scsi.
+ */
+static void
+task_complete_test(void)
+{
+	struct spdk_event event;
+	struct spdk_scsi_task task = {};
+	struct spdk_bdev_io bdev_io = {};
+	struct spdk_scsi_lun lun;
+
+	TAILQ_INIT(&lun.tasks);
+	TAILQ_INSERT_TAIL(&lun.tasks, &task, scsi_link);
+	task.lun = &lun;
+
+	event.arg1 = &task;
+	event.arg2 = &bdev_io;
+
+	task.type = SPDK_SCSI_TASK_TYPE_CMD;
+	bdev_io.status = SPDK_BDEV_IO_STATUS_SUCCESS;
+	spdk_bdev_scsi_task_complete(&event);
+	CU_ASSERT_EQUAL(task.status, SPDK_SCSI_STATUS_GOOD);
+
+	bdev_io.status = SPDK_BDEV_IO_STATUS_FAILED;
+	spdk_bdev_scsi_task_complete(&event);
+	CU_ASSERT_EQUAL(task.status, SPDK_SCSI_STATUS_CHECK_CONDITION);
+	CU_ASSERT_EQUAL(task.sense_data[4], (SPDK_SCSI_SENSE_ABORTED_COMMAND & 0xf));
+	CU_ASSERT_EQUAL(task.sense_data[14], SPDK_SCSI_ASC_NO_ADDITIONAL_SENSE);
+	CU_ASSERT_EQUAL(task.sense_data[15], SPDK_SCSI_ASCQ_CAUSE_NOT_REPORTABLE);
```

Comment:
After reviewing how we use the sense_data array, I suppose the offsets are actually correct - the iSCSI-specific SenseLength is stored in the first two bytes, so all of the offsets are adjusted + 2.

The & 0xf comment still applies, though.